electionleaflets/settings/base.py

```diff
@@ -36,14 +33,64 @@
     root('media'),
 )
 
-DEFAULT_FILE_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
-AWS_S3_FILE_OVERWRITE = False
+if env("AWS_STORAGE_BUCKET_NAME", default=None):
+    import boto.s3.connection
+
+    DEFAULT_FILE_STORAGE = 'storages.backends.s3boto.S3BotoStorage'
+    AWS_S3_FILE_OVERWRITE = env.bool("AWS_S3_FILE_OVERWRITE", default=False)
+    AWS_ACCESS_KEY_ID = env("AWS_ACCESS_KEY_ID")
+    AWS_SECRET_ACCESS_KEY = env("AWS_SECRET_ACCESS_KEY")
+    AWS_STORAGE_BUCKET_NAME = env("AWS_STORAGE_BUCKET_NAME")
+    AWS_S3_HOST = env("AWS_S3_HOST", default="s3-eu-west-1.amazonaws.com")
+    AWS_S3_CUSTOM_DOMAIN = env("AWS_S3_CUSTOM_DOMAIN", default=None)
+    AWS_S3_CALLING_FORMAT = boto.s3.connection.OrdinaryCallingFormat()
+
+if env('MAILGUN_SMTP_SERVER', default=None):
+    EMAIL_HOST = env('MAILGUN_SMTP_SERVER')
+    EMAIL_HOST_USER = env('MAILGUN_SMTP_LOGIN')
+    EMAIL_HOST_PASSWORD = env('MAILGUN_SMTP_PASSWORD')
+    EMAIL_PORT = env('MAILGUN_SMTP_PORT')
+    EMAIL_USE_TLS = True
+    EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+elif env('MAILTRAP_API_TOKEN', default=None):
+    from urllib2 import Request, urlopen
+    import json
+    request = Request('https://mailtrap.io/api/v1/inboxes.json?api_token={}'.format(env('MAILTRAP_API_TOKEN')))
+    response_body = urlopen(request).read()
+    credentials = json.loads(response_body)[0]
+
+    EMAIL_HOST = credentials['domain']
+    EMAIL_HOST_USER = str(credentials['username'])
+    EMAIL_HOST_PASSWORD = str(credentials['password'])
+    EMAIL_PORT = credentials['smtp_ports'][0]
+    EMAIL_USE_TLS = True
+    EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+else:
+    EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
 
 STATICFILES_FINDERS = (
     'django.contrib.staticfiles.finders.FileSystemFinder',
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
 )
 
+DEFAULT_FROM_EMAIL = env('DEFAULT_FROM_EMAIL', default='team@electionleaflets.org')
+EMAIL_RECIPIENT = env('EMAIL_RECIPIENT', default='team@electionleaflets.org')
+REPORT_EMAIL_SUBJECT = env('REPORT_EMAIL_SUBJECT', default='ELECTION LEAFLET REPORT')
+
+MAINTENANCE_MODE = env.bool('MAINTENANCE_MODE', default=False)
+GOOGLE_ANALYTICS_ENABLED = env.bool('GOOGLE_ANALYTICS_ENABLED', default=False)
+
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+ACCOUNT_DEFAULT_HTTP_PROTOCOL = 'https'
+
+TWITTER_KEY = env('TWITTER_KEY', default='')
+TWITTER_SECRET = env('TWITTER_SECRET', default='')
+TWITTER_TOKEN = env('TWITTER_TOKEN', default='')
+TWITTER_TOKEN_SECRET = env('TWITTER_TOKEN_SECRET', default='')
+
+ADMINS = [a.split(',') for a in env('ADMINS', default='Admin,admin@example.com').split(';')]
```

Comment:
Reckon both `,` and `;` are valid in email addresses…but I'm happy committing to not letting people with emails containing them to admin this project ;)